src/scales/scale.time.js

```diff
@@ -50,19 +50,36 @@ module.exports = function(Chart) {
 		return a - b;
 	}
 
-	function buildLookupTable(ticks, min, max, linear) {
-		var ilen = ticks.length;
+	/**
+	 * Returns an array of {time, pos} objects used to interpolate a specific `time` or position
+	 * (`pos`) on the scale, by searching entries before and after the requested value. Since the
+	 * value is linearly interpolated, only timestamps that break the time linearity are added,
+	 * meaning that in the best case, it contains only 2 objects: {min, 0} and {max, 1}. `pos` is
```

Comment:
That doesn't work: `linear` doesn't mean that all ticks are linear but that the lookup table should interpolate `pos` to maintain time linearity (see second image in the screenshot). The check of linearity between 2 points is a great optimization since this code apply to all kind of sources. Currently, in case of `auto`, this table contains only 2 items whatever mode (`linear` or `series`) - the best case can either comes from auto generated ticks or user labels.